js/jquery.mapael.js

```diff
@@ -405,13 +398,36 @@
                 //      Toggle (i.e. click) only if:
                 //          - slice legend is shown AND we want to hide
                 //          - slice legend is hidden AND we want to show
-                $("[data-type='elem']", $(this)).each(function() {
-                    if (($(this).attr('data-hidden') === "0" && showlegendElems === false) || 
-                        ($(this).attr('data-hidden') === "1" && showlegendElems === true)) {
-                        // Toggle state of element by clicking
-                        $(this).trigger('click', [false, animDuration]);
-                    }
-                });
+                if (typeof opt != "undefined" && typeof opt.setLegendElemsState === "object") {
+                    // setLegendElemsState is an object listing the legend we want to hide/show
+                    var $container = $(this); // save this before each for use in callback
```

Comment:
I think you could use the already defined variable $self instead of $container here